addons/website/tests/test_page.py

```diff
@@ -336,3 +338,19 @@ def test_06_alternatives(self):
             self.assertEqual(canonical_url, f'{self.base_url()}/fr/page_1')
             self.assertEqual(alternate_en_url, f'{self.base_url()}/page_1')
             self.assertEqual(alternate_fr_url, f'{self.base_url()}/fr/page_1')
+
+    def test_07_not_authorized(self):
+        # Create page that requires specific user role.
+        specific_page = self.page.copy({'website_id': self.env['website'].get_current_website().id})
+        specific_page.write({
+            'arch': self.page.arch.replace('I am a generic page', 'I am a specific page not available for visitors'),
+            'is_published': True,
+            'visibility': 'restricted_group',
+            'groups_id': [Command.link(self.ref('website.group_website_publisher'))],
+        })
+        # Access page as anonymous visitor.
+        self.authenticate(None, None)
+        r = self.url_open('/page_1')
+        # Check that is is rendered as a website page.
+        self.assertEqual(403, r.status_code, "Must fail with 403")
+        self.assertTrue('id="wrap"' in r.text, "Must be rendered as a website page")
```

Comment:
Note that is exists `assertIn(substring, text)` for that purpose, the advantage of using `assertIn` is that it would print a nice message `f"{substring} was not found in {text}"` instead of `False is not True`. But in this very context I doubt that printing the entire html page in case the text is not found would be of any use, let's keep the `assertTrue` :)